### PR TITLE
cleanup v1.6 LTS

### LINKED ISSUE #166: libgit issues with Julia 1.6.5 h989b2f6_0
There seem to be potential libgit2 issues with Julia 1.6.5 h989b2f6_0

```julia
$ julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.5 (2021-12-19)
 _/ |\__'_|_|_|\__'_|  |  conda-forge-julia release
|__/                   |

(@julia171_test) pkg> activate --temp
  Activating new environment at `/tmp/jl_wZ4SZa/Project.toml`

(jl_wZ4SZa) pkg> add https://github.com/mkitti/libcxxwrap_julia_jll_mock.git#cf
    Updating git-repo `https://github.com/mkitti/libcxxwrap_julia_jll_mock.git`
ERROR: failed to fetch from https://github.com/mkitti/libcxxwrap_julia_jll_mock.git, error: GitError(Code:ERROR, Class:Invalid, invalid version 0 on git_proxy_options)

(jl_wZ4SZa) pkg> add https://github.com/mkitti/libcxxwrap_julia_jll_mock.git
    Updating git-repo `https://github.com/mkitti/libcxxwrap_julia_jll_mock.git`
ERROR: failed to fetch from https://github.com/mkitti/libcxxwrap_julia_jll_mock.git, error: GitError(Code:ERROR, Class:Invalid, invalid version 0 on git_proxy_options)


$ conda list
# packages in environment at /home/mkitti/anaconda3/envs/julia171_test:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
arpack                    3.7.0                hdefa2d7_2    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.18.1               h7f98852_0    conda-forge
ca-certificates           2021.10.8            ha878542_0    conda-forge
curl                      7.80.0               h494985f_1    conda-forge
expat                     2.4.2                h9c3ff4c_0    conda-forge
gettext                   0.19.8.1          h73d1719_1008    conda-forge
git                       2.34.1          pl5321hf874766_0    conda-forge
gmp                       6.2.1                h58526e2_0    conda-forge
julia                     1.6.5                h989b2f6_0    conda-forge
krb5                      1.19.2               h48eae69_3    conda-forge
libblas                   3.9.0           12_linux64_openblas    conda-forge
libcblas                  3.9.0           12_linux64_openblas    conda-forge
libcurl                   7.80.0               h494985f_1    conda-forge
libcxxwrap-julia          0.8.3                h4bd325d_1    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 11.2.0              h1d223b6_11    conda-forge
libgfortran-ng            11.2.0              h69a702a_11    conda-forge
libgfortran5              11.2.0              h5c6108e_11    conda-forge
libgit2                   1.3.0                haabb1ae_1    conda-forge
libgomp                   11.2.0              h1d223b6_11    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
liblapack                 3.9.0           12_linux64_openblas    conda-forge
libnghttp2                1.43.0               ha19adfc_1    conda-forge
libopenblas               0.3.18          pthreads_h8fe5266_0    conda-forge
libopenblas-ilp64         0.3.18          pthreads_h0aa5bea_0    conda-forge
libssh2                   1.10.0               ha35d2d1_2    conda-forge
libstdcxx-ng              11.2.0              he4da1e4_11    conda-forge
libunwind                 1.5.0                h9c3ff4c_1    conda-forge
libutf8proc               2.7.0                h7f98852_0    conda-forge
libzlib                   1.2.11            h36c2ea0_1013    conda-forge
mbedtls                   3.1.0                h9c3ff4c_0    conda-forge
metis                     5.1.0             h58526e2_1006    conda-forge
mpfr                      4.1.0                h9202a9a_1    conda-forge
ncurses                   6.2                  h58526e2_4    conda-forge
openblas-ilp64            0.3.18          pthreads_h3d04fff_0    conda-forge
openlibm                  0.7.0                h516909a_0    conda-forge
openssl                   3.0.0                h7f98852_2    conda-forge
p7zip                     16.02             he1b5a44_1000    conda-forge
pcre2                     10.37                h032f7d1_0    conda-forge
perl                      5.32.1          1_h7f98852_perl5    conda-forge
suitesparse               5.10.1               h9e50725_1    conda-forge
tbb                       2021.5.0             h4bd325d_0    conda-forge
tk                        8.6.11               h27826a3_1    conda-forge
zlib                      1.2.11            h36c2ea0_1013    conda-forge
```

### LINKED ISSUE #167: Set JULIA_SSL_CA_ROOTS_PATH
Julia is not locating the conda-forge ca-certificates.

```julia
julia> using NetworkOptions

julia> NetworkOptions.ca_roots()
"/etc/ssl/certs/ca-certificates.crt"
```

We should set
```
JULIA_SSL_CA_ROOTS_PATH=$CONDA_PREFIX/ssl/cacert.pem
```

then

```
julia> using NetworkOptions

julia> NetworkOptions.ca_roots()
"/home/mkitti/anaconda3/envs/julia171_test/ssl/cacert.pem"

julia> ENV["JULIA_SSL_CA_ROOTS_PATH"]
"/home/mkitti/anaconda3/envs/julia171_test/ssl/cacert.pem"
```

### PR DESCRIPTION
**Pull request by ngam**
Fixes #171 
Fixes #166 
Fixes #167 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
@conda-forge-admin, please restart ci

**Comment by ngam** [comment]
@conda-forge-admin, please restart ci

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

**Comment by ngam** [comment]
Closing this as it we don't want to introduce any breaking changes.

**Comment by mkitti** [comment]
We probably should still merge this, but with a commented out activate.sh as in cfc88fd. This should result in no breakage if someone has already been working with a Julia depot in `~/.julia`. If people want the new depot behavior with 1.6.x, they can uncomment the appropriate lines containing `export`.

@ngam, could you backport the other commits?

**Comment by ngam** [comment]
Sure, we should do that, but let's give people a week or two to weigh in. Copying @zklaus.

**Comment by ngam** [comment]
@mkitti yes I will attend to this soon

File: recipe/build.sh [hunk 1]
```diff
@@ -6,6 +6,9 @@ export LIBRARY_PATH=${PREFIX}/lib
 export CMAKE_PREFIX_PATH=${PREFIX}
 export PATH="${PREFIX}/bin:${PATH}"
 
+#set JULIA_DEPOT_PATH in conda env
+export JULIA_DEPOT_PATH="${PREFIX}/share/julia/site:$JULIA_DEPOT_PATH" 
+
```
Comment (mkitti) [inline recipe/build.sh:11]: This still needs to be removed

File: recipe/build.sh [hunk 2]
```diff
@@ -62,9 +65,18 @@ make -j${CPU_COUNT} prefix=${PREFIX} sysconfigdir=${PREFIX}/etc \
  USE_SYSTEM_ZLIB=1 \
  USE_SYSTEM_P7ZIP=1 \
  ${EXTRA_MAKEFLAGS} \
- TAGGED_RELEASE_BANNER="conda-forge-julia release" \
+ TAGGED_RELEASE_BANNER="A conda-forge release: https://github.com/conda-forge/julia-feedstock" \
```
Comment (mkitti) [inline recipe/build.sh:68]: I notice that wraps in my 80 character wide console. Perhaps we should shorten this somehow? Just include the URL?

Comment (ngam) [reply recipe/build.sh:68]: Yes. I was thinking about the same :)

Comment (ngam) [reply recipe/build.sh]: How about this?

Comment (ngam) [reply recipe/build.sh]: `By conda-forge: github.com/conda-forge/julia-feedstock` = 54

`A conda-forge release: https://github.com/conda-forge/julia-feedstock` = 69

Comment (ngam) [reply recipe/build.sh]: For comparison:

```
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.1 (2021-12-22)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
 _/ |\__'_|_|_|\__'_|  |  By conda-forge: github.com/conda-forge/julia-feedstock
 _/ |\__'_|_|_|\__'_|  |  A conda-forge release: https://github.com/conda-forge/julia-feedstock
 _/ |\__'_|_|_|\__'_|  |  github.com/conda-forge/julia-feedstock

|__/                   |

julia> 

```

File: recipe/build.sh [hunk 3]
```diff
@@ -5,6 +5,10 @@ export LD_LIBRARY_PATH=${PREFIX}/lib
 export LIBRARY_PATH=${PREFIX}/lib
 export CMAKE_PREFIX_PATH=${PREFIX}
 export PATH="${PREFIX}/bin:${PATH}"
+export JULIA_SSL_CA_ROOTS_PATH=$CONDA_PREFIX/ssl/cacert.pem
```
Comment (mkitti) [inline recipe/build.sh:8]: This is needed at runtime via activate.sh, not build time.

File: recipe/build.sh [hunk 4]
```diff
@@ -62,9 +66,18 @@ make -j${CPU_COUNT} prefix=${PREFIX} sysconfigdir=${PREFIX}/etc \
  USE_SYSTEM_ZLIB=1 \
  USE_SYSTEM_P7ZIP=1 \
  ${EXTRA_MAKEFLAGS} \
- TAGGED_RELEASE_BANNER="conda-forge-julia release" \
+ TAGGED_RELEASE_BANNER="By github.com/conda-forge/julia-feedstock" \
```
Comment (mkitti) [inline recipe/build.sh:69]: Does https:// fit here? Without that, my terminal does not recognize the URL.

**Review by mkitti — CHANGES_REQUESTED** [review]
Also this fixes the 1.6 branch. I have not checked what openlibm version Julia 1.6.5 uses.

File: recipe/build.sh (hunk 4 above)
Comment (ngam) [reply recipe/build.sh:69]: > Also this fixes the 1.6 branch. I have not checked what openlibm version Julia 1.6.5 uses.

Should be something in the v0.7.x range. Everything was fine and stable until the v0.8.0 release.

Comment (ngam) [reply recipe/build.sh:69]: > Does https:// fit here? Without that, my terminal does not recognize the URL.

Will change :)

File: recipe/build.sh (hunk 3 above)
Comment (ngam) [reply recipe/build.sh:8]: Ooops sorry

**Comment by ngam** [comment]
> Also this fixes the 1.6 branch. I have not checked what openlibm version Julia 1.6.5 uses.

https://github.com/JuliaLang/julia/blob/96a99c268a96ee61f6ff8fe4c2d53208f0cb042a/deps/Versions.make#L86-L89

**Comment by ngam** [comment]
> > Also this fixes the 1.6 branch. I have not checked what openlibm version Julia 1.6.5 uses.
> 
> https://github.com/JuliaLang/julia/blob/96a99c268a96ee61f6ff8fe4c2d53208f0cb042a/deps/Versions.make#L86-L89
> 
> V0.7.3 --- submitting a build for that. But generally, the differences between the 0.7.x builds are very minor. That's partly why hard pinning is discouraged --- why pin so hard if you can be relax stuff? Upstream should relax their pinnings and give ranges, rather than exact pinning. Oh I forgot ... they vendor it all... 🤣

update: not going to build v0.7.3 --- will require a significant patch to work:

https://github.com/JuliaMath/openlibm/compare/v0.7.3...v0.7.4

**Comment by SylvainCorlay** [comment]
This appears to cause some breakage downstream.

Julia is looking for `PREFIX/lib/julia/libopenlibm.so` (or dylib) at runtime and not finding it.

**Comment by SylvainCorlay** [comment]
Logs of a failing CI build downstream: https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/432647/logs/14

**Comment by SylvainCorlay** [comment]
Thanks, it seems to be fixed with the newest build indeed.

**Comment by SylvainCorlay** [comment]
@mkitti I am seeing the latest build of Julia segfault when doing `Pkg.install`.

https://pipelines.actions.githubusercontent.com/9zddvhQiTJbR8A2Iyvzp5LUMYExviz48Jr6OTWD1xGufbW56YS/_apis/pipelines/1/runs/155/signedlogcontent/3?urlExpires=2022-01-07T12%3A37%3A33.4425814Z&urlSigningMethod=HMACV1&urlSignature=YZJdgN3OwdanhBkm0ksqStE13572V2U2R4OGEmyR1qM%3D

**Comment by mkitti** [comment]
The segfaults are occurring in libcurl ... and 7.81 was just released on conda-forge.

**Pull request merged**